### ISSUE TITLE
Add dataset with alternative identifier

### ISSUE THREAD

**Issue by sdebodt** [ugent-library/biblio-backoffice#116]
**Describe the improvement**
In addition to adding datasets using a DOI, with automated retrieval of metadata using DataCite, users will need to be able to add datasets that do not have a DOI but another persistent identifier or accession number. In this case, no automatic metadata retrieval should be implemented for simplicity reasons at this point. Only validation of the format of the globally unique and persistent identifier is needed. However, we can initially also leave out the validation step and accept any identifier that is provided by the user.

Validation can be done as implemented by the [F-uji tool](https://github.com/pangaea-data-publisher/fuji) (https://www.f-uji.net/index.php)  

[IDutils](https://idutils.readthedocs.io/en/latest/)) validates the following identifiers:
- [globally unique](https://github.com/pangaea-data-publisher/fuji/blob/49c99ef4bdf1a0578dc61bc9e7eb555693887a05/fuji_server/evaluators/fair_evaluator_unique_identifier.py): ID format of ISBN10, ISBN13, ISSN, ISTC, DOI, Handle, EAN8, EAN13, ISNI ORCID, ARK, PURL, LSID, URN, Bibcode, arXiv, PubMed ID, PubMed Central ID, GND, SRA, BioProject, BioSample, Ensembl, UniProt, RefSeq, Genome Assembly.
- [persistent](https://github.com/pangaea-data-publisher/fuji/blob/49c99ef4bdf1a0578dc61bc9e7eb555693887a05/fuji_server/evaluators/fair_evaluator_persistent_identifier.py): ark', 'arxiv', 'bioproject', 'biosample', 'doi', 'ensembl', 'genome', 'gnd', 'handle', 'lsid', 'pmid', 'pmcid', 'purl', 'refseq', 'sra', 'uniprot', 'urn'

**Expected behavior**
An identifier is provided by the user in the "Add reference" step. Next (after validation), the user can proceed to the "Complete description" step to manually provide the metadata fields for this dataset.

Instead of enumerating all possible types of identifiers as listed above, allow the user to choose between DOI and "alternative persistent identifier" (dropdown).

Alternative identifiers should be entered as urls that can be resolved to landing pages or as urls of the landing pages themselves (see examples below).

**Screenshots**
![image](https://user-images.githubusercontent.com/72742841/142230450-b35cce19-d5f0-43d7-827a-39c5226ebc1a.png)

**Related context**
examples of alternative identifiers: 
- http://hdl.handle.net/10934/RM0001.COLLECT.5223
- https://www.ebi.ac.uk/ena/browser/view/LR796137
- https://hdl.handle.net/21.11115/0000-000C-E12A-7
- https://ega-archive.org/datasets/EGAD00000000001

**Comment by sdebodt** [comment]
Investigate which metadata harvesters can be used for the above identifiers.
Currently, the approach is not developed to provide metadata without import.

**Comment by sdebodt** [comment]
- make epic from this issue and add smaller, specific issues.
- data curators can help to identify appropriate methods
- ideally, to postpone until LibreCat reimplementation is further progressed. Timing to be checked.

**Comment by mietcls** [comment]
We'll pick this one up after the launch, we need to thoroughly discuss with the data curators what is expected and needed to get the right datasets. As I understood it's a pretty big ticket, with implications on the data quality.

**Comment by sdebodt** [comment]
> * consider to have a manual import workflow (cf publication) for datasets with alternative persistent identifiers

Can we discuss whether this manual workflow is an option in case automated import based on alternative identifiers would be postponed until mid 2023 or later?

**Comment by mietcls** [comment]
@sdebodt could you describe
- When is the manual import needed, in which cases, why? How often does this happen?
- Why don't these datasets have a DOI? Will they get a DOI later?
- What is needed for a manual import? How is it different from an automatic import?

If I understand correctly, it's possible that duplicate information will arise. If a dataset does not have a DOI, and there is an other version _with_ a DOI, we'll have to figure out which one is the right version (there are some identifiers that might help, but this is added complexity for the future that we'll need to flesh out before we make this decision). Also, what do we do when a dataset is later on published with a DOI? Is this correct, am I missing anything @nics?

If we'll decide to go from manual to automatic at some point; is this something we'll have to replace completely (and we'll have to clean up), or is it functionality we can extend and iterate upon functionally? How will we go from manual to non-manual? Can we postpone it to make sure we don't have to do double work if needed?

If you're here tomorrow/Wednesday before noon by any chance, I can hop in and discuss.

**Comment by sdebodt** [comment]
@mietcls this concerns datasets that do not have a DOI, but have another persistent identifier. Some data repositories (often domain-specific) use other persistent identifiers than DOIs (https://rdm.elixir-belgium.org/identifiers).

This means that this dataset or a subsequent version will not receive a DOI. 

The manual approach would mean that we offer the user with an empty form (identical to the one we currently have) with an ID field for which the ID type is specified.

As with the manual import for publications, datasets that are manually created would not need adaptations when later on automatic import would be possible.

The automatic import requires to call different services (than the DataCite service) to retrieve metadata based on the PID. We judged that the implementation of this automatic import would best be done once the LibreCat replacement is done, to avoid having to develop it twice.

**Comment by nics** [comment]
@mietcls @sdebodt replaced by #898
